armbalancer.go

```diff
@@ -72,16 +72,43 @@ func New(opts Options) http.RoundTripper {
 	}
 
 	if opts.TransportFactory == nil {
-		opts.TransportFactory = newRecyclableTransport
+		opts.TransportFactory = make(map[string]func(id int, parent *http.Transport, host string, port string, recycleThreshold, minReqsBeforeRecycle int64) http.RoundTripper)
+		opts.TransportFactory[strings.ToLower(host+":"+port)] = newRecyclableTransport
 	}
 
-	t := &transportPool{pool: make([]http.RoundTripper, opts.PoolSize)}
-	for i := range t.pool {
-		t.pool[i] = newRecyclableTransport(i, opts.Transport, host, port, opts.RecycleThreshold, opts.MinReqsBeforeRecycle)
+	t := &pooledTransport{pool: make(map[string]*transportPool)}
+	for key, factory := range opts.TransportFactory {
+		transports := make([]http.RoundTripper, 0, opts.PoolSize)
+		for i := 0; i < opts.PoolSize; i++ {
+			transports = append(transports, factory(i, opts.Transport, host, port, opts.RecycleThreshold, opts.MinReqsBeforeRecycle))
+			t.pool[key] = &transportPool{pool: transports}
+		}
 	}
 	return t
 }
 
+type pooledTransport struct {
```

Comment:
it isn't immediately obvious what the responsibilities of `pooledTransport` are relative to `transportPool`. Would recommend renaming to something like `hostScopedTransport` for clarity.